src/object/schema_manager.c

```diff
@@ -15707,6 +15696,22 @@ sm_collect_truncatable_classes (MOP class_mop, std::unordered_set < OID > &trun_
 	      return er_errid ();
 	    }
 
+	  int partition_type = DB_NOT_PARTITIONED_CLASS;
+	  error = sm_partitioned_class_type (class_mop, &partition_type, NULL, NULL);
```

Comment:
I think you want to check fk_child_mop, not class_mop.

Reply:
super thank you!